apps/judicial-system/web/src/routes/Shared/DetentionRequests/DetentionRequests.tsx

```diff
@@ -157,6 +160,41 @@ export const DetentionRequests: React.FC = () => {
     return sortConfig.key === name ? sortConfig.direction : undefined
   }
 
+  const deleteCase = async (caseToDelete: Case) => {
+    if (
+      caseToDelete.state === CaseState.NEW ||
+      caseToDelete.state === CaseState.DRAFT
+    ) {
+      const transitionRequest = parseTransition(
+        caseToDelete.modified,
+        CaseTransition.DELETE,
+      )
+
+      try {
+        const { data } = await transitionCaseMutation({
+          variables: { input: { id: caseToDelete.id, ...transitionRequest } },
+        })
+        if (!data) {
+          return
+        }
+
+        setRequestToRemoveIndex(undefined)
+
+        setTimeout(() => {
+          setCases(
+            cases?.filter((c: Case) => {
+              return c !== caseToDelete
+            }),
+          )
+        }, 800)
```

Comment:
Perhaps we should add timeouts to all queries and mutations.
But isn't a timeout of less than a second for a server request that updates the database a bit aggressive?